sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -224,9 +224,10 @@ namespace Azure { namespace Core { namespace Http {
     Azure::Core::IO::BodyStream* GetBodyStream() { return this->m_bodyStream; }
 
     /**
-     * @brief A value indicating whether download is happening via stream.
+     * @brief A value indicating whether download will fill a memory buffer or if it will provide a
+     * body stream instead.
      */
-    bool IsDownloadViaStream() { return this->m_isDownloadViaStream; }
+    bool IsNotBufferDownload() { return this->m_isDownloadViaStream; }
```

Comment:
How about ShouldDownloadViaStream?

```suggestion
    bool ShouldDownloadViaStream() { return this->m_isDownloadViaStream; }
```

Reply:
IsBufferedDownload() ??

Reply:
I like that too, but it inverts the bool. And using buffered download is the common case. We want the common case to have the default value of the bool to be false.

Reply:
Well, I thought we wanted to get rid of `stream` from the name and make it all related to buffer :) 

@RickWinter  changing to `IsBufferedDownload()` changes the meaning and would require inverting the logic everywhere.  But I think is worth it.